### ISSUE TITLE
ServerEndpointConfig connection leak (remains open) if connection is aborted

### ISSUE THREAD

**Issue by szarza** [eclipse-ee4j/glassfish#23985]
## Environment Details
* GlassFish Version (and build number): 5.1
* JDK version: 8
* OS: Linux

----------

## Problem Description
Websocket endpoint connection remains open even a ServerEndpointConfig.Configurator.modifyHandshake() has been overridden to abort connection. In WSServer.java, if the line 

~~~
@ServerEndpoint(value = "/events/all", configurator = WSServerConfigurator2.class)
~~~

is used, the connection is aborted on the client-side, but not on the server-side, which remains open. Depending on whatever does the server endpoint, that could generate also a memory leak

## Steps to reproduce
Attached demonstrator project which includes a file video recording demonstrating that connections are still opened although are closed on the client, showing how to reproduce the issue and including how to find the leak in a heap dump.

[Please, see the related bug in the Tyrus project with the demonstrator project and video](https://github.com/eclipse-ee4j/tyrus/issues/791)

**Comment by jansupol** [comment]
In the case of the GF 5.1. branch is to be released, Tyrus 1.19 provides the fix.

**Comment by szarza** [comment]
Thanks @jansupol
